hibernate-core/src/main/java/org/hibernate/dialect/HSQLDialect.java

```diff
@@ -223,15 +219,6 @@ public SqlAstTranslatorFactory getSqlAstTranslatorFactory() {
 		};
 	}
 
-	@Override
-	public String castPattern(CastType from, CastType to) {
-		if ( from== BOOLEAN
-				&& (to== INTEGER || to== LONG)) {
-			return "casewhen(?1,1,0)";
```

Comment:
Again, it's kinda sad to lose this nice terse SQL syntax, and replace it with a verbose `case`/`when`.

Reply:
Passing null would produce the else-result, so this wouldn't be compatible with other implementations. Thus it needs case when.

Reply:
Actually HSQL has a `decode()` function, so we could use that, like in Oracle.

Reply:
You are right, thanks. I'll add that.